Update copyright years

eht16 opened the pull request:
As mentioned in #2030.

elextr commented:
Technically anywhere that copyright is mentioned should say contributors instead of listing a few people since Geany has no copyright assignment and everyone who has ever contributed still owns copyright of their contribution.

eht16 commented:
I'm not sure if "contributors" is enough for legal requirements. I'm not a lawyer, still I assume using an undefined bunch of people (i.e. "contributors") properly is not valid for copyright assignment.

@elextr are you referring just to the About dialog or any copyright notice in the whole source code?

Maybe someone else has a better knowledge of the legal requirements.
@frlan ?

elextr commented:
> I'm not sure if "contributors" is enough for legal requirements. I'm not a lawyer, still I assume using an undefined bunch of people (i.e. "contributors") properly is not valid for copyright assignment.

Well, "contributors" as recorded in the VCS record is what I meant.  That is definitely recorded for contributions since git, and in the prior SVN commit comments where committers were fairly diligent in acknowledging contributors.

> are you referring just to the About dialog or any copyright notice in the whole source code?

Oh, I didn't think of that, I was only thinking about things that are visible like the dialogs.

IANAL as you know, but I was alerted to it by problems that another project had when someone re-used their (substantial) contribution under another license and the project objected.  The contributor pointed out they owned the material and could license it any way they wanted and if the project continued to object they would sue it for claiming copyright of material it didn't own because it had a notice very similar to Geanys.  The project is now offline, although I can't say if it is because of that issue.

codebrainz commented:
As far as I know - and this may just be a North American thing - but it's not required to explicitly indicate copyright using such notices, but rather the act of publishing something automatically grants the copyright. If this is indeed the case, the mailing list/version control history provides the proof of copyright ownership of the original authors.

Alternatively, we could require contributors to transfer copyright to the Geany foundation (or whatever it's called) which is setup in Germany. Personally I'm against this as it provides too much friction and I personally never contribute to projects which require this.

elextr commented:
My understanding matches @codebrainz and IIUC its the case in a lot of jurisdictions, not just NA.

> Alternatively, we could require contributors to transfer copyright to the Geany foundation (or whatever it's called) which is setup in Germany.

Agree this is the wrong way to go, and (not meaning to be difficult of course :) I would not agree to transfer copyright of my existing contributions.

codebrainz commented:
Other than doing nothing, the easiest solution would be to just append [", et al."](https://en.wiktionary.org/wiki/et_al.) at the end of each copyright notice line, probably.

codebrainz commented:
Also if the year (range) is replaced with "2005-Present", it would avoid having to update these each release … however, IANAL.

frlan commented:
In Europe is nearly clear: You cannot get lost of your Urheberrecht. You don't have to explicit claim it., Which is slightly different from American copyright. However as Geany is a product of it's community I suggest to make it more general.

eht16 commented:
Ok, so in the source files I would replace
```Copyright 2005-2012 Enrico Tröger <enrico(dot)troeger(at)uvena(dot)de>``` by ```Copyright 2005-present Geany developers and contributors``` and similar in the About dialog?
Or any better suggestions?

While at it, I would also remove the "All rights reserved" notice as it seems not necessary at all nowaydays (according to Wikipedia).

codebrainz commented:
> Or any better suggestions?

Maybe remove the end date (present) altogether, I don't think it changes when the copyright will expire (long after us).

Possibly just leave the old names/emails and append ", et al." or "and others".

[For example this](https://gist.github.com/codebrainz/5c9447c7e36d4e33cbde0dd90853383c/revisions?diff=split). I don't have any real preference though.

>While at it, I would also remove the "All rights reserved" notice as it seems not necessary at all nowaydays (according to Wikipedia).

👍 I don't think it's consistently used either.

elextr commented:
Agree with @codebrainz that we don't need a year range or `all rights reserved`, see the "how to apply" in the GPL itself.

Agree with @eht16 that it should say "contributors", its not any old person, its those who are listed by the VCS as contributing.  Its important in legal issues to make a reasonable attempt at precision, just hand waving "and others" isn't appropriate.

codebrainz commented:
>  Its important in legal issues to make a reasonable attempt at precision, just hand waving "and others" isn't appropriate.

I'm don't know Latin, and am not a lawyer, but I believe that ["et al."](https://dictionary.cambridge.org/dictionary/english/et-al) is legalese for this exact use. Also, it's technically more precise to leave some of the names and abbreviate it with "et al." rather than deleting the original details and replacing with some ambiguous group like "Geany developers" or "contributors". But again, I don't really care.

elextr commented:
The term "contributors" at least is a term commonly understood in the open source development world.

"Geany Developers" is a defined list and does not include most contributors, so is inappropriate.

eht16 commented:
After reading some more about the topic, I tend to use one
of the following:

`Copyright 20XX The Geany authors and contributors`

`Copyright 20XX The Geany contributors`

with replacing the year with the first year in the
existing copyright notices (i.e. use the first "publication").
I tried to find good and authoritative examples of projects which
already solved this but only found a wide variety of different
copyright notices.

What I actually would like to remove are the names we currently have
in the source files. I wasn't aware this is not strictly necessary
legally and so never thought about it.
As you stated, the detailed authorship of a file and even single code
lines can be queried much more appropriate by the VCS.

https://opensource.com/law/14/2/copyright-statements-source-files has a
nice explanation why author names are not a good idea in copyright
notices: false implications by contributors thinking this is the
authors' code or they need to add themselves for proper crediting
and maybe more.

elextr commented:
@eht16, sounds good to me, and nice reference.

codebrainz commented:
> Copyright 20XX The Geany contributors

This one seems fine.

eht16 commented:
Here we go, I:
- replaced individual copyright holders by "The Geany contributors"
- removed everything but the initial copyright year
- removed copyright notice and author names from README as the copyright is not necessary here and author names (and their email addresses) tend to outdate and be incomplete
- skipped manually updating the translation files as the copyright string of the About dialog changed too much this time (I cannot translate "The Geany contributors" properly to zh_CN or hi :)
- skipped anything in `ctags` and `scintilla`
- am sorry for the huge diff :(

elextr commented:
@eht16 checked all but the po files by inspection, apart from the occasional trailing space which was also removed this appears to do only what it says.

For the po files, shouldn't they wait until the coming release since they would be regenerated then anyway?

eht16 commented:
I have the po files regenerated already. What I meant in my comment was that now the translations for the copyright notice string in the About dialog is missing because it changed a lot and I skipped to guess the translations for all the different languages I do not know :).
So, in short: there is nothing to do since the translators will probably update it within the next string freeze.

elextr commented:
> I have the po files regenerated already.

I was more worried about having two merges of the PO files in quick succession.

And to be clear the rest looks ok to merge.

eht16 commented:
> two merges of the PO files in quick succession

The updates in this PR should be safe, I mean, it's GIT :). And I intend to squash a bit anyway.
In case you refer to the upcoming string freeze updates, I'm not concerned at all because I think we will merge this one before (I'd like just to wait for feedback from @codebrainz  and/or @b4n). Even if, usually po files are easy to handle.

b4n commented on src/main.c:
```diff
@@ -1,7 +1,7 @@
  *      main.c - this file is part of Geany, a fast and lightweight IDE
  *
- *      Copyright 2014 Matthew Brush <mbrush@codebrainz.ca>
+ *      Copyright 2005 The Geany contributors
```
That date change is kinda odd

eht16 replied:
This is because: https://github.com/geany/geany/commit/d33758da926ca124db674240c431b8b07bb0e683#diff-803c5170888b8642f2a97e5e9423d399L5. `src/main.c` was one of the very first files ever.

b4n replied:
yeah but the old main.c became the new libmain.c, didn't it?  anyway, it doesn't matter, and yes the file was there and did contain something reasonably similar.

b4n commented on src/tagmanager/tm_parser.h:
```diff
@@ -1,6 +1,6 @@
 /*
 *
-*   Copyright (c) 2014, Colomban Wendling
+*   Copyright (c) 2005 The Geany contributors
```
Here as well, although I'm surprised the content of this file is actually new since 2014, I'd probably have been moved here then instead, so maybe 2005 is fine indeed.

eht16 replied:
Good catch, this was actually wrong as the file was first introduced in 2014 in https://github.com/geany/geany/commit/6a85a50d21b6297fe1ea51ecc8792f6a1f81cf58#diff-55056236371f88550f26cb87580ec0a4. Sorry, fixed.

eht16 commented:
@b4n you are right about the unnecessary PO files regeneration. Will remember the next time.
For now, I would keep it for not complicating the process any more and they'll get regenerated soon anyway for the upcoming string freeze.

If there will be no other remarks, I would squash the last three commits into one and merge soon.
